org.opent2t.sample.lamp.superpopular/com.hue.bulb/js/colour.js

```diff
@@ -0,0 +1,359 @@
+"use strict";
+
+const ChangeTolerance = 0.0001;
+const MaxHue = 360.0;
+const MaxColour = 255;
+
+/*
+ * Method to see if the given XY value is within the reach of the lamps.
+ *
+ * @param xyColour is a two-element object representing a point containing the X,Y value
+ * @param colourGamut   the color gamut of the light device.
+ * @return true if within reach, false otherwise.
+ */
+function isPointInLampsReach(xyColour, colourGamut){
```

Comment:
>xyColour [](start = 29, length = 8)

Nit: For consistency do we go with the british variant or n.American (color vs colour?) I am not sure what we have been doing elsewhere but I've typically stuck to NA patterns in other projects..